src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandWrapper.cs

```diff
@@ -62,7 +62,7 @@ public IEdmTypeReference GetEdmType()
             }
 
             Type elementType = GetElementType();
-            return model.GetEdmTypeReference(elementType);
+            return model.GetTypeMappingCache().GetEdmType(elementType, model);
```

Comment:
>GetTypeMappingCache [](start = 25, length = 19)

Inside GetTypeMappingCache, we call this every time - typeMappingCache = model.GetAnnotationValue<ClrTypeCache>(model); is this a cheap enough call? Should we cache the ClrTypeCache somewhere and may be pass it as an argument?